pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -2242,9 +2242,24 @@ func (ds *DistSender) sendToReplicas(
 				}
 
 				if ds.kvInterceptor != nil {
-					numReplicas := len(desc.Replicas().Descriptors())
-					reqInfo := tenantcostmodel.MakeRequestInfo(ba, numReplicas)
-					respInfo := tenantcostmodel.MakeResponseInfo(br, !reqInfo.IsWrite())
+					var reqInfo tenantcostmodel.RequestInfo
```

Comment:
I liked your original implementation better. I don't think splitting the read and write paths saved an allocation. reqInfo and respInfo are flat structs with no embedded pointers.